sdk/metric/periodic_reader.go

```diff
@@ -120,6 +120,10 @@ func NewPeriodicReader(exporter Exporter, options ...PeriodicReaderOption) Reade
 		flushCh:  make(chan chan error),
 		cancel:   cancel,
 		done:     make(chan struct{}),
+		rmPool: sync.Pool{
```

Comment:
Should this be scoped to only _a_ periodic reader? Moving this to the package or file scope will allow for the amortization the memory allocations across a larger use.

Reply:
Yes, it might allow use across multiple Periodic readers, but you then will probably not get reuse benefits.  It is more likely that your reader will have about the same shape from collect to collect, but that different readers might not have the same metrics in the same order.

Now I would support using the pool across all Collect calls, both periodic and manual, in which I think having a shared pool for all readers could make sense.  I know there was pushback on that initially.

Reply:
> might allow use across multiple Periodic readers, but you then will probably not get reuse benefits

It looks like we reuse the allocated `ScopeMetrics` and `Metrics` slices. I'm not sure I follow how these will differ significantly across periodic readers in the common case. I'd expect both readers to export for the same scopes and each to have some metrics. Is there something I'm missing with the common use case here?

Given a pool is intended to reduce allocations to the heap, the chance of having a `ScopeMetric` or `Metric` slice allocated once and then that allocation reused increases when the pool is shared across more periodic readers.

There are is also overhead to a `sync.Pool` having one per periodic reader reduces that overhead.

> I would support using the pool across all Collect calls

Wouldn't this leak the pool resources to the external caller of collect?

Reply:
We can address this in a later PR if needed.